src/aws_credentials_file.erl

```diff
@@ -106,9 +127,20 @@ parse_credentials_file(Path, Options) ->
 parse_config_file(Path, Options) ->
     {ok, F} = file:read_file(Path),
     {ok, Profiles} = eini:parse(F),
-    Desired = maps:get(profile, Options, <<"default">>),
+    Desired = desired_profile(Options),
+    read_from_profile(Profiles, Desired).
+
+-spec read_from_profile(map(), binary()) -> any().
+read_from_profile(File, Profile) ->
+    case maps:get(Profile, File, undefined) of
+        undefined -> {error, {desired_profile_not_found, Profile}};
+        Map -> {ok, Map}
+    end.
 
-    case maps:get(Desired, Profiles, false) of
-        false -> {error, {desired_profile_not_found, Desired}};
-        Profile -> {ok, Profile}
+-spec desired_profile(aws_credentials_provider:options()) -> binary().
+desired_profile(Options) ->
+    case {os:getenv("AWS_PROFILE"), maps:get(profile, Options, undefined)} of
+        {false, undefined} -> <<"default">>;
+        {false, Profile} -> Profile;
+        {AwsProfile, undefined} -> list_to_binary(AwsProfile)
```

Comment:
Hmmm, considering the way we handle `HOME` and `credential_path`, I used the same behaviour, we should use the `Options` first and then try to get from env. Because if we always choose from env, users wouldn't be able to change it without changing the env var.

WDYT?

Reply:
Makes sense to me :+1: So `env` should be the "last choice" here in that case :+1: